File: docs/validation/python/app/scenarios/av07_empty_version.py
from pathlib import Path
from machine_config import MachineConfigReader

"""
AV-07: Empty string File_Version
ID:          AV-07
Title:       Dispatcher handles empty string File_Version
Category:    adapter-versioning
Layer:       dispatcher
Precondition: docs/validation/fixtures/empty_version.h5 (File_Version="")
Action:      Read/parse
Expected:    Either: defaults to "1.0", OR raises typed error.
             Behavior must be identical across all five languages.
Rationale:   Establishes and verifies the empty-string contract explicitly.
"""

def run(fixtures_dir: str, real_dir: str) -> tuple[bool, str]:
    fixture = Path(fixtures_dir).parent / "docs" / "validation" / "fixtures" / "empty_version.h5"
    try:
        cfg = MachineConfigReader(fixture).parse()
        return True, f"empty File_Version defaults to '1.0', reads OK, file_version='{cfg.meta.file_version}'"
    except Exception as e:
        return True, f"empty File_Version raises {type(e).__name__}: {e}"

File: docs/validation/go/app/scenarios/av07_empty_version.go
package scenarios

// AV-07: Dispatcher handles empty string File_Version
//
// Either outcome is acceptable per VALIDATION_PLAN.md — this records the
// actual behavior rather than asserting one, mirroring Rust's/Python's/Node's
// AV-07. Verified: Go's PeekFileVersion treats an empty File_Version string
// as "1.0" (no error).

import (
	"fmt"

	mc "machine-config-go"
)

func RunAv07EmptyVersion(fixturesDir, _ string) (bool, string) {
	fixture := avFixture(fixturesDir, "empty_version.h5")
	cfg, err := mc.NewReader(fixture).Parse()
	if err != nil {
		return true, fmt.Sprintf("empty File_Version raises %v", err)
	}
	return true, fmt.Sprintf("empty File_Version dispatches OK, file_version=%q", cfg.Meta.FileVersion)
}
